src/prefect/records/result_store.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+from typing import Any
+
+from prefect.exceptions import ObjectNotFound
+from prefect.results import BaseResult, PersistedResult, ResultFactory
+from prefect.utilities.asyncutils import run_coro_as_sync
+
+from .store import RecordStore
+
+
+@dataclass
+class ResultFactoryStore(RecordStore):
+    result_factory: ResultFactory
```

Comment:
Doesn't need to be part of this PR, but I think it'd be ideal if we could have some default behavior for `result_factory` and improve the UX for configuration. I took a stab at creating a `ResultFactory` during testing, and it is not easy.

Reply:
Yes!! I also was feeling that intensely during this work; for example, there's no pattern for creating a `PersistedResult(...)` class and then saving an object as a separate method call.  You _have_ to go through the full `create` mechanics which always optimistically writes to the `path` every time.